cli/genpolicy/genpolicy.go

```diff
@@ -51,14 +51,22 @@ func New(rulesPath, settingsPath, cachePath string, bin []byte) (*Runner, error)
 // Run runs the tool on the given yaml.
 //
 // Run can be called more than once.
-func (r *Runner) Run(ctx context.Context, yamlPath string, logger *slog.Logger) error {
+func (r *Runner) Run(ctx context.Context, yamlPath string, cmPath []string, logger *slog.Logger) error {
 	args := []string{
 		"--runtime-class-names=contrast-cc",
 		"--rego-rules-path=" + r.rulesPath,
 		"--json-settings-path=" + r.settingsPath,
 		"--layers-cache-file-path=" + r.cachePath,
 		"--yaml-file=" + yamlPath,
 	}
+
+	if len(cmPath) > 0 {
+		// "-c" is shared between the kata and microsoft genpolicy for different long flags
+		args = append(args, "-c="+cmPath[0])
+	} else if len(cmPath) > 1 {
+		return fmt.Errorf("multiple files with ConfigMaps found, only one is allowed")
+	}
```

Comment:
This is a limitation that needs to be documented somewhere.

Reply:
This is actually a regression that I'd like to avoid. Maybe we could join the yamls into a temp file?

Reply:
Actually, It's even worse since the current PR already breaks of the file in cmPath contains more than one YAML document (or that one document isn't a ConfigMap) since the upstream genpolicy implementation requires this: https://github.com/kata-containers/kata-containers/blob/fdc0d81198ad556d019e8a9dbb6b2aeb5ecf8fef/src/tools/genpolicy/src/policy.rs#L458